src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Extensions/TextDocumentExtensions.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.Extensions;
+internal static class TextDocumentExtensions
+{
+    /// <summary>
+    /// Mutates the <see cref="TextDocumentIdentifier"/> by changing the <see cref="Uri"/>. Returns it again for convenience.
+    /// </summary>
+    public static TextDocumentIdentifier WithUri(this TextDocumentIdentifier textDocumentIdentifier, Uri uri)
```

Comment:
I don't love this but it did make something easier to write. Thoughts?

Reply:
Mutation is the only way to do this, lest we lose something, so I can see why the method makes sense. Not sure if we have a different naming convention for methods that mutatin things